docs/src/operations/upgrade.md

```diff
@@ -20,11 +20,117 @@ The following procedure describes how to upgrade a broker. If the cluster contai
+
+  * If you're using Kubernetes 1.17+, then there is a new beta feature, which you can use to snapshot your pvc. Check [this documentation](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) for more information.
+  * If you're using an older Kubernetes version then you need to create snapshot's of your pvc's with help of your cloud provider.  For example for the google cloud kubernetes engine follow [this guide](https://cloud.google.com/compute/docs/disks/create-snapshots).
+
+#### Performing the Upgrade
+
+If you using helm you can update the image tag in your values file.
+
+```yaml
+image:
+  repository: camunda/zeebe
+  tag: NEW_VERSION_TAG
+```
+
+After you did this you can run `helm upgrade` to roll out the new version.
+
+```shell script
+ helm upgrade RELEASE_NAME zeebe/zeebe-cluster -f VALUES_FILE_NAME.yaml
+```
+
+#### Verifying the Upgrade
+
+The upgrade is successful if the following conditions are met:
+
+* the broker's are ready (see [Ready Check](/operations/health.md#ready-check))
+* the broker's are healthy (see [Health Check](/operations/health.md#health-check))
+* all partitions are healthy (see the [Metric](/operations/metrics.md#metrics-related-to-health) `zeebe_health`)
+* the stream processors of the partition leaders are in the phase `PROCESSING` (see [Partitions Admin Endpoint](#partitions-admin-endpoint))
+
+If the upgrade failed because of a known issue, then a partition change its status to unhealthy, and the log output may contain the following error message:
+
+<details>
+  <summary>Sample Upgrade Error Message</summary>
+  <p>
+
+```
+Unexpected error on recovery happens.
+io.zeebe.engine.processor.InconsistentReprocessingException: Reprocessing issue detected!
+  Restore the data from a backup and follow the recommended upgrade procedure. [cause:
+  "The key of the record on the log stream doesn't match to the record from reprocessing.",
+  log-stream-record: {"partitionId":1,"value":{"version":1,"bpmnProcessId":"parallel-tasks",
+  "workflowKey":2251799813685249,"parentElementInstanceKey":-1,"parentWorkflowInstanceKey":-1,
+  "bpmnElementType":"PARALLEL_GATEWAY","flowScopeKey":2251799813685251,
+  "elementId":"ExclusiveGateway_0tkgnd5","workflowInstanceKey":2251799813685251},
+  "key":2251799813685256,"sourceRecordPosition":4294997784,"valueType":"WORKFLOW_INSTANCE",
+  "timestamp":1601025180728,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING",
+  "rejectionType":"NULL_VAL","rejectionReason":"","position":4294998112},
+  reprocessing-record: {key=2251799813685255, sourceRecordPosition=4294997784,
+  intent=WorkflowInstanceIntent:ELEMENT_ACTIVATING, recordType=EVENT}]
+```
+
+  </p>
+</details>
+
+In this case, the broker should be rolled back to the previous version and the backup should be restored. Ensure that the upgrade was prepared correctly. If it is still unclear why it was not successful then please contact the Zeebe team and ask for guidance.
```

Comment:
I think for standard dev users, it's not clear how they would rollback to a previous version and use the backup to restore from. But I don't know if we want to go in details here...maybe at least link to the relevant docs how to use a `VolumeSnapshot` (e.g. link directly to the relevant paragraph here https://kubernetes.io/docs/concepts/storage/volume-snapshots/#provisioning-volumes-from-snapshots). Is this something we can do with the helm chart though? I don't think you can specify the volumes or any info. correct me if im wrong @salaboy

Reply:
I did in line 35 and 36. Actually I would like to avoid these details :sweat_smile: The guides from the official docs are far better probably. And if we want to have this kind of guide then we need to do it by ourself which we haven't done as far as I know.

Reply:
Makes sense :slightly_smiling_face: I'll defer to @salaboy if users can figure this out from just the links in the docs.